alvr/server/cpp/platform/linux/Renderer.hpp

```diff
@@ -0,0 +1,127 @@
+#pragma once
+
+#define VULKAN_HPP_NO_CONSTRUCTORS
+#include <vulkan/vulkan.h>
+#include <vulkan/vulkan.hpp>
+#include <vulkan/vulkan_core.h>
+#include <vulkan/vulkan_enums.hpp>
+#include <vulkan/vulkan_handles.hpp>
+#include <vulkan/vulkan_structs.hpp>
+
+#include "VkContext.hpp"
+#include "utils.hpp"
+
+#include "ffmpeg_helper.h"
+#include "monado_interface.h"
+
+namespace alvr::render {
+
+constexpr u32 ImageCount = 6;
+constexpr usize StagingImgCount = 2;
+
+enum class HandleType {
+    None,
+    DmaBuf,
+    OpaqueFd,
+};
+
+struct Image {
+    vk::Image image = VK_NULL_HANDLE;
+    vk::ImageLayout layout = vk::ImageLayout::eUndefined;
+    vk::DeviceMemory memory = VK_NULL_HANDLE;
+    vk::ImageView view = VK_NULL_HANDLE;
+
+    void destroy(VkContext const& ctx)
+    {
+        ctx.dev.destroy(view);
+        ctx.dev.free(memory);
+        ctx.dev.destroy(image);
+    }
+};
+
+struct Output {
+    Image image;
+    DrmImage drm;
+    // VkSemaphore semaphore;
+    VkImageCreateInfo imageCI;
+    VkDeviceSize size;
+};
+
+struct PipelineCreateInfo {
+    std::vector<u8> shaderData;
+    std::vector<vk::SpecializationMapEntry> specs;
+    std::vector<u8> specData;
+};
+
+struct RendererCreateInfo {
+    vk::Format format;
+    vk::Extent2D inputEyeExtent;
+    vk::Extent2D outputExtent;
+    std::array<int, ImageCount> inputImgFds;
+};
+
+namespace detail {
+
+    class RenderPipeline {
+        vk::ShaderModule shader;
+        vk::PipelineLayout pipeLayout;
+        vk::Pipeline pipe;
+
+    public:
+        RenderPipeline(VkContext const& ctx, vk::DescriptorSetLayout& layout, PipelineCreateInfo& pipelineCI);
+
+        void render(
+            VkContext const& ctx, vk::CommandBuffer cmdBuf, vk::ImageView in, vk::ImageView out, vk::Extent2D outSize);
+
+        void destroy(VkContext const& ctx);
+    };
+
+}
+
+class Renderer {
+    vk::Extent2D eyeExtent;
+    vk::Extent2D outExtent;
+
+    Image inputImages[ImageCount];
+    Image stagingImgs[StagingImgCount];
+    Output output;
+
+    vk::QueryPool timestampPool;
+    vk::CommandPool cmdPool;
+    vk::CommandBuffer cmdBuf;
+    vk::Sampler sampler;
+    vk::DescriptorSetLayout descLayout;
+    vk::Fence fence;
+
+    std::vector<detail::RenderPipeline> pipes;
+
+    // vk::Semaphore monadoFinishedSem;
```

Comment:
Monado stuff that's unused should be removed , of course make sure you have it in another branch we don't want you losing work(if there's more/more important stuff I haven't caught yet)

Reply:
I'll rename it and keep it because I expect them to actually add sync to the present, so this is essentially a workaround comment out